woocommerce/woocommerce-ios
# Analytics: Integrate ExPlat into Experiments framework

Closes: #5621

## Description

This integrates the Experimentation Platform (ExPlat) service from the Tracks library, so we can run ExPlat A/B tests. See the [Tracks library documentation](https://github.com/Automattic/Automattic-Tracks-iOS#to-run-experiments) for how ExPlat experiments are run.

The ExPlat service will be started or refreshed when the app is started and the user is logged in, or after logging in.

## Changes

* Updates `TracksProvider` to send the WordPress.com auth token to Tracks, so Tracks can configure ExPlat for authenticated users.
* Adds `Tracks` to the `Experiments` framework, so we can do the A/B test setup from that framework.
* Adds an `ABTest` enum to `Experiments`:
   * ExPlat experiments can be added as cases to that enum, with experiment variations defined from the ExPlat service.
   * The ExPlat service is started whenever there is more than one experiment (i.e. more than just the `null` case), by registering the experiments with ExPlat.
* Starts the A/B testing for authenticated users in `AppDelegate` and after login in `StorePickerViewController`.
* Updates the docs with details for how to add an A/B test and define test and control behavior.

## Testing

There are no experiments running yet, but you can confirm that the ExPlat service starts as expected:

1. Add a breakpoint at the beginning of the `start()` method in `ABTest`.
2. Run `bundle exec pod install`.
3. Build and run the app.
4. When you are already logged in to the app, confirm `ABTests.start()` is hit when the app launches but doesn't register any experiments (since there are none to register for now).
5. Log out and log in, and confirm `ABTests.start()` is hit when you finish logging in (when the login epilogue screen appears).

## Submitter Checklist

Update release notes:
